release-team/role-handbooks/ci-signal/upgrade-job-versions.md

```diff
@@ -0,0 +1,137 @@
+# Upgrade Job Versions
+
+An example of how, given an upgrade job, the different ways I can verify:
+- what version is it upgrading FROM (old-version)
+- what version is it upgrading TO (new-version)
+
+## TL;DR
+- almost no jobs actually use a released/tagged build 
+- most jobs use one of these versions, built off the HEAD of a branch
+
+| name           | branch eg    | actual eg                           |
+| -------------- | ------------ | ----------------------------------- |
+| ci/latest      | master       | v1.13.0-alpha.0.1226+426ef9d349bb3a |
+| ci/k8s-beta    | release-1.12 | v1.12.0-beta.1.129+1d58f1aebfe1e3   |
+| ci/k8s-stable1 | release-1.11 | v1.11.3-beta.0.71+a4529464e4629c    |
+| ci/k8s-stable2 | release-1.10 | v1.10.8-beta.0.33+a963fce72fed31    |
+| ci/k8s-stable3 | release-1.9  | v1.9.11-beta.0.36+06bf888123ca89    |
+
+NB: these examples are based on the current cycle at time of writing, which is release-1.12
+
+### A: Look at the job name
+aka trust the job
+
+ref: https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md#test-jobs-naming-convention
+
+- upgrade-cluster: master + nodes upgraded to new-version, old-version tests run
+- upgrade-cluster-new: master + nodes upgraded to new-version, new-version tests run
+- upgrade-master: master upgraded to new-version, nodes left at old-version, old-version tests run
+- TODO: do any jobs use a tagged release
```

Comment:
we also have suites like https://k8s-testgrid.appspot.com/sig-release-master-upgrade#gce-new-master-upgrade-cluster-parallel

which only runs parallel tests.. maybe also worth mention?

Reply:
also jobs like https://k8s-testgrid.appspot.com/sig-release-1.11-all#kubeadm-gce-upgrade-1-10-1-11 is using a tagged release

Reply:
I'm pretty much just trying to constraint this to upgrade versions, it could be that longer term the e2e test docs need a rewrite but I wrote this with a specific case in mind

Reply:
hum, those jobs use `release/latest`, is there any specific reason they should be using that and  not `release/stable`

Reply:
that kubeadm job is an upgrade job :-) although looks like you've mentioned it towards bottom of this doc.

Reply:
> hum, those jobs use release/latest, is there any specific reason they should be using that and not release/stable

unfortunately we (as test-infra) don't maintain those job configs... @timothysc probably can help redirect to whomever currently maintains kubeadm jobs in cluster-lifecycle?